tests/test_smartspim.py

```diff
@@ -0,0 +1,272 @@
+"""
+Tests the SmartSPIM acquisition metadata creation
+"""
+
+import copy
+import os
+import shutil
+import tempfile
+import unittest
+from unittest.mock import MagicMock, patch
+
+from aind_data_schema.core import acquisition
+
+from aind_metadata_mapper.smartspim.acquisition import (
+    JobSettings,
+    SmartspimETL,
+)
+
+from .resources.smartspim.example_metadata import (
+    example_filter_mapping,
+    example_metadata_info,
+    example_processing_manifest,
+    example_session_end_time,
+)
+
+
+class TestSmartspimETL(unittest.TestCase):
+    """Tests methods in the SmartSPIM class"""
+
+    def setUp(self):
+        """Setting up temporary folder directory"""
+        self.temp_dir = tempfile.mkdtemp()
```

Comment:
I'm really hoping to avoid creating folders and writing operations in the unit tests